### ISSUE TITLE
navigable positions in ReplicaSDK's coordinate frame 

### ISSUE THREAD

**Issue by iszihan** [facebookresearch/Replica-Dataset#29]
Hello, I'm trying to get navigable positions for replica dataset with habitat-sim.  It seems habitat-sim's coordinate system is quite different from the one used in ReplicaSDK (the gravity direction is in +y as opposed to -z?).  I am wondering if there is a known transformation between these two coordinate systems? Much thanks in advance.

**Comment by erikwijmans** [comment]
Given an agents position `pos` and rotation `rot`,
```python
# Define the transforms needed
_tmp = np.eye(3, dtype=np.float32)
UnitX = _tmp[0]
UnitY = _tmp[1]
UnitZ = _tmp[2]

R_hsim_replica = habitat_sim.utils.quat_from_two_vectors(-UnitZ, -UnitY)
T_hsim_replica = np.eye(4, dtype=np.float32)
T_hsim_replica[0:3, 0:3] = quaternion.as_rotation_matrix(R_hsim_replica)

R_forward_back = habitat_sim.utils.quat_from_two_vectors(-UnitZ, UnitZ)
T_replicac_hsimc = np.eye(4, dtype=np.float32)
T_replicac_hsimc[0:3, 0:3] = quaternion.as_rotation_matrix(R_forward_back)


# Transforming the state happens here
T_hsim_c = np.eye(4, dtype=np.float32)
T_hsim_c[0:3, 0:3] = quaternion.as_rotation_matrix(rot)
T_hsim_c[0:3, 3] = pos

T_c_hsim = np.linalg.inv(T_hsim_c)
T_c_replica = T_replicac_hsimc @ T_c_hsim @ T_hsim_replica

T_replica_c = np.linalg.inv(T_c_replica)

replica_pos = T_replica_c[0:3, 3]
replica_rot = quaternion.from_rotation_matrix(T_replica_c[0:3, 0:3])
```

There are two things that are different in the ReplicaSDK vs. habitat-sim: The gravity direction and in Replica the camera is RDF while habitat-sim uses RUB.

**Comment by iszihan** [comment]
@erikwijmans  this is perfect!! thank you! :)

**Comment by iszihan** [comment]
@erikwijmans  looked the code more closely (it works perfect) and I'm just a little confused on one detail, and would very appreciate it if there is an answer: what does the transformation of 'T_replicac_hsimc' do (the one rotates -UnitZ to +UnitZ)? I removed 'T_replicac_hsimc' from the matrix multiplication to 'T_c_replica =  T_c_hsim @ T_hsim_replica' and the transformed position I got seem to be the same. Is there a specific reason to multiply 'T_replicac_hsimc' on the left side ?

Thank you! :)
